src/python_testing/matter_testing_support.py

```diff
@@ -1008,9 +1008,11 @@ def skip_all_remaining_steps(self, starting_step):
                 test is more deliberately identifying where test skips are starting from, making
                 it easier to validate against the test plan for correctness.
         '''
-        last_step = len(self.get_test_steps(self.current_test_info.name)) + 1
-        for index in range(starting_step, last_step):
-            self.skip_step(index)
+        steps = self.get_test_steps(self.current_test_info.name)
+        starting_step = [idx for idx, step in enumerate(steps) if step.test_plan_number == starting_step][0]
+        remaining = steps[starting_step:]
+        for step in remaining:
+            self.skip_step(step.test_plan_number)
```

Comment:
I think something like this would provide a more helpful error when an incorrect `starting_step_number` is provided
```suggestion
        for idx, step in enumerate(steps):
                if step.test_plan_number == starting_step:
                        starting_step_idx = idx
                        break
        else:
                raise TestingExpection("skip_all_remaining_steps was provided with invalid starting_step_num")
        remaining = steps[starting_step_idx:]
        for step in remaining:
            self.skip_step(step.test_plan_number)
```